### PR TITLE
Unit improvements

### PR DESCRIPTION
**Pull request by pascalpp**
- adds a Cycle Units menu command with shortcut `U`
- adds unit labels in the ruler for `PX`, `MM`, and `IN`
- improves mouse tick drawing code

image

image

image

image

### CODE REVIEW

File: Free Ruler/de.lproj/MainMenu.strings [hunk 1]
```diff
@@ -133,3 +133,6 @@
 
 /* Class = "NSMenuItem"; title = "Inches"; ObjectID = "lt1-Hj-2TR"; */
 "lt1-Hj-2TR.title" = "Inch";
+
+/* Class = "NSMenuItem"; title = "Cycle Units"; ObjectID = "2nm-aL-kZd"; */
+"2nm-aL-kZd.title" = "Einheiten Durchlaufen";
```
Comment (pascalpp) [inline Free Ruler/de.lproj/MainMenu.strings:138]: @J-rg This PR adds a menu item to cycle through the units pixels, millimeters, and inches. Can you review this localized string?

File: Free Ruler/fi.lproj/MainMenu.strings [hunk 2]
```diff
@@ -133,3 +133,6 @@
 
 /* Class = "NSMenuItem"; title = "Inches"; ObjectID = "lt1-Hj-2TR"; */
 "lt1-Hj-2TR.title" = "Tuumat";
+
+/* Class = "NSMenuItem"; title = "Cycle Units"; ObjectID = "2nm-aL-kZd"; */
+"2nm-aL-kZd.title" = "Vaihda Yksiköitä";
```
Comment (pascalpp) [inline Free Ruler/fi.lproj/MainMenu.strings:138]: @lanurmi This PR adds a menu item to cycle through the units pixels, millimeters, and inches. Can you review this localized string?

File: Free Ruler/Base.lproj/MainMenu.xib [hunk 3]
```diff
@@ -238,11 +246,5 @@
             </items>
             <point key="canvasLocation" x="-147" y="125"/>
         </menu>
-        <menuItem title="Group Rulers" state="on" keyEquivalent="g" id="7SO-Lm-Ylr">
-            <modifierMask key="keyEquivalentModifierMask"/>
-            <connections>
-                <action selector="toggleGroupRulers:" target="Voe-Tx-rLC" id="RkN-cz-hMe"/>
-            </connections>
-        </menuItem>
```
Comment (pascalpp) [inline Free Ruler/Base.lproj/MainMenu.xib:246]: this was a weird stray menuitem that somehow got added outside the menu, not visible in the app. A duplicate of the existing Group Rulers menu item.

File: Free Ruler/fi.lproj/MainMenu.strings (hunk 2 above)
Comment (lanurmi) [inline Free Ruler/fi.lproj/MainMenu.strings:138]: Should be: "Vaihda yksikköä"

File: Free Ruler/de.lproj/MainMenu.strings (hunk 1 above)
Comment (J-rg) [reply Free Ruler/de.lproj/MainMenu.strings:138]: The D should not be capitalized.
Though I feel this does not clearly convey what the menu item does; a non-literal translation such as „Nächste Einheit auswählen“ („Select Next Unit“) might be a more intelligible title.

Also, „Inch“ in the line above should be changed to „Zoll“.

**Comment by pascalpp** [comment]
@jeff-h Do you have some time to try out this PR before I merge it out?

**Comment by pascalpp** [comment]
@carlfriess Do you have some time to try out this PR before I merge it out?

File: Free Ruler/RuleView.swift [hunk 4]
```diff
@@ -56,6 +64,28 @@ class RuleView: NSView {
...
+    func getUnitLabel() -> String {
+        switch prefs.unit {
+        case .pixels:
+            return "PX"
+        case .millimeters:
+            return "MM"
+        case .inches:
+            return "IN"
+        }
+    }
```
Comment (carlfriess) [inline Free Ruler/RuleView.swift:76]: Maybe use lowercase spelling? Personally, I think it looks slightly better.

```suggestion
    func getUnitLabel() -> String {
        switch prefs.unit {
        case .pixels:
            return "px"
        case .millimeters:
            return "mm"
        case .inches:
            return "in"
        }
    }
```

**Comment by carlfriess** [comment]
Looks good to me! :)

**Comment by jeff-h** [comment]
Hey, sorry I haven't had a chance to look at anything properly recently — just been flat out with other work. Appreciate all your work though! These latest features are awesome.

**Comment by pascalpp** [comment]
@jeff-h no worries!

**Pull request merged**